drivers/charger/charger_pf1550.c

```diff
@@ -0,0 +1,695 @@
+/*
+ * Copyright 2024 Arduino SA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_pf1550_charger
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/charger.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/sys/linear_range.h>
+
+#include "zephyr/logging/log.h"
+LOG_MODULE_REGISTER(pf1550_charger);
```

Comment:
use `LOG_MODULE_REGISTER(pf1550_charger, CONFIG_CHARGER_LOG_LEVEL);`, this way you can control the log level with the subsystem option